City-of-Helsinki/drupal-module-helfi-tpr
# UHF-10352

# [UHF-10352](https://helsinkisolutionoffice.atlassian.net/browse/UHF-10352) Delete TPR entities from UI

## What was done

Allow all 5 TPR entities to be deleted from UI

## How to install
* Make sure your instance is up and running on latest dev branch.
    * `git pull origin dev`
    * `make fresh`
* Update the Helfi Platform config and Helfi TPR
    * `composer require drupal/helfi_platform_config:dev-UHF-10352`
    * `composer require drupal/helfi_tpr:dev-UHF-10352`
* Run `make drush-updb drush-cr`

## How to test

- Admin, Content producer and editor should have updated permissions as shown in[ platform_config PR](https://github.com/City-of-Helsinki/drupal-helfi-platform-config/pull/879/files)
- Login as Admin, Content producer or editor
- Go to `/admin/content/integrations/tpr-unit`
  - You should be able to delete any unit that no longer exists in the TPR-API 
- Go to `/admin/content/integrations/tpr-service`
  - You should be able to delete any service that no longer exists in the TPR-API
- Go to `/admin/content/integrations/tpr-errand-service`
  - You should be able to delete any errand service that no longer exists in the TPR-API

Trying to delete an entity which can be found from api shows a message to user.

[UHF-10352]: https://helsinkisolutionoffice.atlassian.net/browse/UHF-10352?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
